channelflow/nse.cpp

```diff
@@ -809,39 +809,74 @@ Real viscosity(Real Reynolds, VelocityScale vscale, MeanConstraint constraint, R
     return nu;
 }
 
+// logic
+//
+// Vsuck == 0
+//   pressure constraint         =>   regular PCF/PPF formula, pressure style
+//   bulk velocity constraint    =>   regular PCF/PPF formula, bulk vel style
+// Vsuck != 0
+//   pressure constraint
+//     dPdx == 0                 =>   regular ASBL formula
+//     dPdx != 0                 =>   pressure ASBL formula
+//   bulk velocity constraint    =>   bulk velocity ASBL formula
+
 ChebyCoeff laminarProfile(Real nu, MeanConstraint constraint, Real dPdx, Real Ubulk, Real Vsuck, Real a, Real b,
                           Real ua, Real ub, int Ny) {
     ChebyCoeff u(Ny, a, b, Spectral);
 
-    if (constraint == BulkVelocity) {
-        if (abs(Vsuck) > 1e-14)
-            cferror("Combining suction with constraint BulkVelocity is not implemented yet");
-
-        u[0] = 0.125 * (ub + ua) + 0.75 * Ubulk;
-        u[1] = 0.5 * (ub - ua);
-        u[2] = 0.375 * (ub + ua) - 0.75 * Ubulk;
-
-    } else {
-        if (abs(Vsuck) < 1e-14) {
+    if (Vsuck == 0.0) {
```

Comment:
This if statement is very dangerous. Compare two real number is not machine precision safe.